tests/optimizers/bohb_test.py

```diff
@@ -82,6 +87,36 @@ def test_bohb_parallel():
         assert es.optimizer_info["configuration_key"] == (1, 0, i)
 
 
+def test_bohb_report_as_batch():
+    paramspace = ps.ParameterSpace()
+    paramspace.add(ps.ContinuousParameter("p1", [0, 1]))
+    opt = BOHB(
+        paramspace,
+        Objective("loss", False),
+        min_fidelity=0.2,
+        max_fidelity=1,
+        num_iterations=1,
+    )
+
+    evaluations = []
+    for i in range(3):
+        es = opt.get_evaluation_specification()
+        evaluation = es.create_evaluation(objectives={"loss": i})
+        evaluations.append(evaluation)
+
+    # Add an evaluation _not_ created by the optimizer to see if it get's skipped
+    # and triggers a warning on reporting:
```

Comment:
Let's be more specific here and say that it's an evaluation that is missing specific details in the optimizer info, because we might well add support for reporting evaluations for custom specifications independent from this test.